runelite-client/src/main/java/net/runelite/client/plugins/objectindicators/ObjectIndicatorsPlugin.java

```diff
@@ -466,11 +471,21 @@ private void savePoints(final int id, final Set<ObjectPoint> points)
 		Set<ObjectPoint> points = GSON.fromJson(json, new TypeToken<Set<ObjectPoint>>()
 		{
 		}.getType());
+
 		// Prior to multiloc support the plugin would mark objects named "null", which breaks
 		// in most cases due to the specific object being identified being ambiguous, so remove
 		// them
+		// .map() necessary for added color to points
 		return points.stream()
 			.filter(point -> !point.getName().equals("null"))
+			.map(point ->
```

Comment:
I assume this should be changed to match the behavior of the ground markers plugin. In fact, this whole PR should probably match the behavior which is present there. That is:

* A config for "remember marker colors" should be added
* When selected, the color used should be the color stored on the point
  * If that color is null (from existing config prior to this enhancement), use the current config color
* If deselected, all markers will use the current config color (current behavior)

As part of that change, this block will no longer be necessary--null color handling will then be done in the overlay.

Reply:
I like that. Good stuff man. I'll work on adding logic to overlay so that the null logic can be added there. by taking out this, then all "previously" marked points and any future marked points marked while config.("remember marker colors").equals(false), will have a null color for point. 

the save would then not over-ride this, and to get a specific color, a point would need to be un-marked and re-marked while the new config is true.

The new config would default to false, so no-one has any change in behavior unless they actively opt-in to using the new config.